pkg/network/node/node.go

```diff
@@ -311,38 +311,44 @@ func (node *OsdnNode) validateMTU() error {
 		return fmt.Errorf("unable to determine MTU while performing validation")
 	}
 
-	if int(node.networkInfo.MTU) > mtu+50 {
+	needsTaint := mtu < int(node.networkInfo.MTU)+50
+	if needsTaint {
 		resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			klog.V(2).Infof("Configured node MTU is more than default interface MTU plus VXLAN overhead, tainting node...")
-			n, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
-
+			nodeObj, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
 			if err != nil {
 				return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
 			}
-
-			oldNode, err := json.Marshal(n)
-
-			if err != nil {
-				return fmt.Errorf("could not marshal old Node object: %v", err)
-			}
-
-			nodeClone := n.DeepCopy()
-
 			const MTUTaintKey string = "network.openshift.io/mtu-too-small"
-			taintedNode, _, err := taints.AddOrUpdateTaint(nodeClone, &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"})
+			mtuTooSmallTaint := &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"}
+			tainted := taints.TaintExists(nodeObj.Spec.Taints, mtuTooSmallTaint)
+			nodeObjCopy := nodeObj.DeepCopy()
+			var nodeWithTaintMods *corev1.Node
+
+			if tainted {
+				klog.V(2).Infof("Default interface MTU is less than VXLAN overhead, tainting node...")
+				nodeWithTaintMods, _, err = taints.AddOrUpdateTaint(nodeObjCopy, mtuTooSmallTaint)
+				if err != nil {
+					return fmt.Errorf("could not taint the node with key %s: %v", MTUTaintKey, err)
+				}
+			} else {
+				klog.V(2).Infof("Node has too small MTU taint but default interface MTU is big enough, untainting node...")
```

Comment:
This is behind the `if needsTaint {` conditional, so I don't think it will actually untaint the node.. The logic is more like
```
if needsTaint && !tainted {
    AddTaint()
} else if !needsTaint && tainted {
    RemoveTaint()
}
```